src/main/java/seedu/duke/utils/SessionManager.java

```diff
@@ -1,24 +1,31 @@
 package seedu.duke.utils;
 
 import seedu.duke.objects.Inventory;
+import seedu.duke.types.Types;
 
 public class SessionManager {
-    private static SessionManager sessionManager = null;
+    private static boolean isAutoSaveActive = true;
 
-    private SessionManager() {
-    }
-    public static SessionManager getInstance() {
-        if (sessionManager == null) {
-            sessionManager = new SessionManager();
-        }
-        return sessionManager;
+    public SessionManager() {
     }
 
     public static void writeSession(Inventory inventory) {
-        Storage.writeCSV(inventory);
+            Storage.writeCSV(inventory);
     }
 
     public static Inventory getSession() {
         return Storage.readCSV();
     }
+
+    public static boolean getAutoSave() {
+        return isAutoSaveActive;
+    }
+
+    public static void setAutoSave(boolean AutoSaveMode) {
+        isAutoSaveActive = AutoSaveMode;
```

Comment:
boolean name AutoSaveMode does not fit coding convention (just make it isAutoSaveActive or something)